src/Avalonia.FreeDesktop/DBusSystemDialog.cs

```diff
@@ -113,35 +131,38 @@ public override async Task<IReadOnlyList<IStorageFile>> OpenFilePickerAsync(File
                     if (x.Results.TryGetValue("current_filter", out var currentFilter))
                     {
                         var name = currentFilter.GetItem(0).GetString();
-                        selectedType = new FilePickerFileType(name);
                         var patterns = new List<string>();
                         var mimeTypes = new List<string>();
                         var types = currentFilter.GetItem(1).GetArray<VariantValue>();
-                        foreach(var t in types)
+                        foreach (var t in types)
                         {
                             if (t.GetItem(0).GetUInt32() == 1)
                                 mimeTypes.Add(t.GetItem(1).GetString());
                             else
                                 patterns.Add(t.GetItem(1).GetString());
                         }
-
-                        selectedType.Patterns = patterns;
-                        selectedType.MimeTypes = mimeTypes;
+                        
+                        // Reuse the file type objects from options
+                        // so the consuming code can match exactly the
+                        // file type selected instead of spawning one.
+                        selectedType = options.FileTypeChoices?.FirstOrDefault(type => type.Name == name ||
+                            (type.MimeTypes?.All(y => mimeTypes.Contains(y)) ?? false) ||
+                            (type.Patterns?.All(y => patterns.Contains(y)) ?? false));
```

Comment:
We probably should change `nameCondition || mimeCondition || patternCondition` into `nameCondition && (mimeCondition || patternCondition)`